### ISSUE TITLE
Create App: "null value in column "device type" violates not-null constraint" on DB container

### ISSUE THREAD

**Issue by openedhardware** [balena-io/open-balena#95]
Version: 3.0.1
Balena CLI verion: 12.26.1

Executed command:
```
balena app create yokosoApp --debug --type raspberry-pi
```

Error log on DB container:
```
2020-11-02 23:59:12.343 UTC [181] ERROR:  null value in column "device type" violates not-null constraint
2020-11-02 23:59:12.343 UTC [181] DETAIL:  Failing row contains (2020-11-02 23:59:12.339564, 9, 24, 1, yokosoApp, yokosoapp, null, null, 0, 1, null, 0, 2020-11-02 23:59:12.339564, 0, 1, 404db469f27f4c0cb63989d33d49aeea).
2020-11-02 23:59:12.343 UTC [181] STATEMENT:  INSERT INTO "application" ("actor", "should track latest release", "organization", "app name", "slug", "is for-device type", "application type", "uuid")
        SELECT "application"."actor", "application"."should track latest release", "application"."organization", "application"."app name", "application"."slug", "application"."is for-device type", "application"."application type", "application"."uuid"
        FROM (
                SELECT "application"."created at", "application"."modified at", "application"."id", "application"."actor", "application"."should track latest release", "application"."organization", "application"."app name", "application"."slug", "application"."is for-device type", "application"."should be running-release", "application"."depends on-application", "application"."application type", "application"."is host", "application"."is archived", "application"."uuid"
                FROM (
                        SELECT CAST(NULL AS TIMESTAMP) AS "created at", CAST(NULL AS TIMESTAMP) AS "modified at", CAST(NULL AS INTEGER) AS "id", CAST($1 AS INTEGER) AS "actor", CAST($2 AS INTEGER) AS "should track latest release", CAST($3 AS INTEGER) AS "organization", CAST($4 AS TEXT) AS "app name", CAST($5 AS VARCHAR(255)) AS "slug", CAST($6 AS INTEGER) AS "is for-device type", CAST(NULL AS INTEGER) AS "should be running-release", CAST(NULL AS INTEGER) AS "depends on-application", CAST($7 AS INTEGER) AS "application type", CAST(NULL AS INTEGER) AS "is host", CAST(NULL AS INTEGER) AS "is archived", CAST($8 AS TEXT) AS "uuid"
                ) AS "application"
        ) AS "application" RETURNING "id";
```

Detailed video: https://u.pcloud.link/publink/show?code=XZ1UxHXZ2EMx05jIcwRRTlBD1LvR64NoTkOX

**Comment by bartversluijs** [comment]
This is probably related to #94. The same error occurs when provisioning a device.

**Comment by sradevski** [comment]
@openedhardware I just created a new OB instance and there were no issues. I suspect this is due to a migration not executed correctly, and the table still has `device_type` column (which should be `is_for__device_type` now), @dfunckt does that sound plausible?

**Comment by openedhardware** [comment]
@sradevski 
Hmm, I had deleted all docker image files and pulled again with `/bin/compose up` command.

Let me try again.

Thanks!

**Comment by dfunckt** [comment]
Let's move the discussion over to https://github.com/balena-io/open-balena/issues/94
